app/post.go

```diff
@@ -812,25 +812,21 @@ func (a *App) parseAndFetchChannelIdByNameFromInFilter(channelName, userId, team
 }
 
 func (a *App) searchPostsInTeam(teamId string, userId string, paramsList []*model.SearchParams, modifierFun func(*model.SearchParams)) (*model.PostList, *model.AppError) {
-	channels := []store.StoreChannel{}
```

Comment:
This is not the simplest to migrate.

The thing is, we are migrating the store to be "Sync" by default, but using explicit async where is needed. In this case is needed. We were calling the `Search` method for each parameter in parallel. So... we define a list of channels, in this cases it should be something like `channels := []chan store.StoreResult`, and then generate each channel in the for loop in an async way using this approach here: https://github.com/mattermost/mattermost-server/pull/10613/files#diff-b3db6e589754eb0bf1d4e2b8af8d9d30R588

And the rest of the could will keep unchanged, I think.